service/controller/resource/prometheusremotewrite/create.go

```diff
@@ -66,3 +55,82 @@ func updateMeta(c, d metav1.Object) {
 	d.SetDeletionGracePeriodSeconds(c.GetDeletionGracePeriodSeconds())
 	d.SetManagedFields(c.GetManagedFields())
 }
+
+func (r *Resource) setRemoteWrite(ctx context.Context, remoteWrite *pmov1alpha1.RemoteWrite, prometheus *promv1.Prometheus) error {
+	err := r.ensureStatusCreated(ctx, remoteWrite, prometheus)
```

Comment:
I'm guessing to put this after the update of Prometheus CR, as the last call in this function.
To ensure that the Prometheus has been updated.

Reply:
We would end up having a status for Prometheus that has failed to configure the remotewrite for some reason.